api/lib/application/cache/cache-controller.js

```diff
@@ -3,9 +3,9 @@ const _ = require('lodash');
 
 module.exports = {
 
-  refreshCacheEntries() {
-    return Promise.all(_.map(AirtableDatasources, (datasource) => datasource.refreshAirtableCacheRecords()))
-      .then(() => null);
+  refreshCacheEntries(request, h) {
+    _.forEach(AirtableDatasources, (datasource) => datasource.refreshAirtableCacheRecords());
+    return h.response().code(204);
   },
```

Comment:
Je préfèrerais qu'on garde le `Promise.all` qui permet d'avoir quand même une bonne gestion d'erreurs. Pour pouvoir retourner une réponse au client sans attendre, il faudrait déplacer cet appel dans une fonction asynchrone dont on n'attendra pas le résultat.

Quelque chose comme (non testé) :

```suggestion
  refreshCacheEntries(request, h) {
	async function _backgroundRefreshCacheEntries() {
	  try {
        await Promise.all(_.map(AirtableDatasources, (datasource) => datasource.refreshAirtableCacheRecords()));
      } catch (e) {
        logger.error("Error while reloading cache", e);
      }
	}
	_backgroundRefreshCacheEntries();
    return h.response().code(202);
  },
```

Reply:
Que pense-tu de cela (code non testé) pour alléger le code du try/catch:

```js
refreshCacheEntries(request, h) {
 function _backgroundRefreshCacheEntries() {
   Promise.all(_.map(AirtableDatasources, (datasource) => 
    datasource.refreshAirtableCacheRecords();
   ).catch((e) {
     logger.error("Error while reloading cache", e);
   });
  }
  _backgroundRefreshCacheEntries();
  return h.response().code(202);
},
```

Reply:
Dans ce cas je dirais même que le fait de stocker dans une fonction n'a pas d'importance ? 

```js
refreshCacheEntries(request, h) {
  Promise.all(_.map(AirtableDatasources, (datasource) => 
    datasource.refreshAirtableCacheRecords();
  ).catch((e) {
    logger.error("Error while reloading cache", e);
  });
  return h.response().code(202);
},
```

Reply:
Ça peut marcher aussi. J'essayais de respecter la règle d'utiliser systématiquement `async`/`await` mais c'est peut-être plus simple comme ça.

Notez que le `Promise.all` n'a plus d'effet dans la proposition de @frinyvonnick, parce que le `catch` est mis au niveau de chaque appel — donc on aura plusieurs logs s'il y a plusieurs erreurs mais ce n'est pas plus mal en fait. Du coup on peut revenir à :

```js
  refreshCacheEntries(request, h) {
    AirtableDatasources.forEach((datasource) => {
      datasource.refreshAirtableCacheRecords().catch((e)=>{
        logger.error("Error while reloading cache", e);
      })
    });
    return h.response().code(202);
  },
```

Et en mode `for/try/catch` pour comparaison ça donnerait :

```js
  refreshCacheEntries(request, h) {
    for(const datasource of AirtableDatasources) {
      try {
        datasource.refreshAirtableCacheRecords();
      } catch(e) {
        logger.error("Error while reloading cache", e);
      }
    }
    return h.response().code(202);
  },
```

Reply:
@jonathanperret j'ai testé rapidement mon code dans la console du navigateur et le catch est bien appelé qu'une seule fois 🤔 Je me suis trompé quelque part ?

![image](https://user-images.githubusercontent.com/13099512/78872006-df407980-7a48-11ea-8d92-77a26dcd7fb3.png)

Reply:
@frinyvonnick En fait la proposition de @celineung était structurée comme suit :

```js
Promise.all(_.map(ds, (d) => d.refresh();).catch((e) { log(e); });
```

Si on enlève le `;` superflu et qu'on ajoute le `=>` manquant pour que ce soit syntaxiquement correct :

```js
Promise.all( _.map(ds, (d) => d.refresh()).catch((e) => { log(e); }) );
```

Mais c'est encore incorrect parce qu'on appelle `.catch()` sur le résultat de `_.map` ce qui ne marche pas.

On peut corriger ça soit en remettant le `.catch()` en dehors (en l'appelant sur le résultat de `Promise.all()` donc, c'est ce que tu as fait dans ta console), soit en le remontant pour l'appeler sur le résultat de `refresh()` (avec l'avantage mentionné de logger une erreur par datasource), ce qui donne :

```js
Promise.all(_.map(ds, (d) => d.refresh().catch((e) => { log(e); }));
```

Et c'est là que du coup le `Promise.all()` n'a aucun effet, du coup la sortie du `map` est superflue, et donc on en vient à :

```js
ds.forEach( (d) => d.refresh().catch((e) => { log(e); }) );
```

J'espère que c'est plus clair !